### PR TITLE
AG-13491: Add `depthOptions` support to grouped-category

### PR DESCRIPTION
**Pull request by iMoses**
https://ag-grid.atlassian.net/browse/AG-13491
https://ag-grid.atlassian.net/browse/AG-13509

### CODE REVIEW

File: packages/ag-charts-types/src/chart/axisOptions.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ export interface AgBaseAxisLabelStyleOptions {
     /** The font family to use for the labels */
     fontFamily?: FontFamily;
     /** Padding in pixels between the axis label and the tick. */
-    padding?: PixelSize;
+    spacing?: PixelSize;
```
Comment (alantreadway) [inline packages/ag-charts-types/src/chart/axisOptions.ts:127]: Please document breaking changes in the `Breaking Changes` field of the related JIRA, here and below.

File: packages/ag-charts-community/src/chart/axis/groupedCategoryAxis.test.ts
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/axis/groupedCategoryAxis.test.ts:1]: Should we add some tests to cover the new `depthOptions` cases?

File: packages/ag-charts-types/src/chart/axisOptions.ts (hunk 1 above)
Comment (iMoses) [reply packages/ag-charts-types/src/chart/axisOptions.ts:127]: Added a separate ticket to track this change:
https://ag-grid.atlassian.net/browse/AG-13509

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**